app/models/container_group.rb

```diff
@@ -94,4 +94,18 @@ def disconnect_inv
     self.deleted_on = Time.now.utc
     save
   end
+
+  def self.post_refresh_ems(ems_id, _ems_refresh_start_time)
+    self.reconnect_events(ems_id)
+  end
+
+  def self.reconnect_events(ems_id)
+    events = EmsEvent.where(:ems_id => ems_id, :container_group_id => nil).where.not(:container_group_name => nil)
+    events.each do |event|
+      event_data = event.full_data
+      next unless event_data[:kind] == "Pod"
+      event.container_group = ContainerGroup.where(:ems_ref => event_data[:uid]).last
```

Comment:
s/where(x).last/find_by(x)/ ?

Reply:
Some of our pod events are also related to containers:
https://github.com/ManageIQ/manageiq-providers-kubernetes/blob/ef6039818308c25baab8abadaf124aa28952e108/app/models/manageiq/providers/kubernetes/container_manager/event_catcher_mixin.rb#L94

Do we need to connect them to containers as well?

Reply:
> s/where(x).last/find_by(x)/ ?

right, I always have this fear that uids are recycled and that it might connect to an old pod

> Do we need to connect them to containers as well?

Seems those events dont get attached to containers but rather to their pod. solving that is the second half of https://bugzilla.redhat.com/show_bug.cgi?id=1496179